### ISSUE TITLE
Querying an unmapped property leads to MySQLSyntaxErrorException

### ISSUE THREAD

**Issue by rnavarropiris** [tomatophantastico/sparqlmap-core#1]
# Problem description

It appears that querying a property which was not defined in the r2rml mapping leads to a weird exception.

The expected behaviour would be the generation of an empty result set, 
# Exception

```
Problem: com.mysql.jdbc.exceptions.jdbc4.MySQLSyntaxErrorException: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'WHERE (CAST((CASE WHEN NULL IS NULL AND NULL IS NULL THEN CAST('true' AS CHAR) E' at line 1
2015-08-03 13:36:15 [http-nio-9090-exec-5] ERROR o.a.c.c.C.[.[.[.[dispatcherServlet] - Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception [Request processing failed; nested exception is com.hp.hpl.jena.query.QueryException: java.sql.SQLException: com.mysql.jdbc.exceptions.jdbc4.MySQLSyntaxErrorException: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'WHERE (CAST((CASE WHEN NULL IS NULL AND NULL IS NULL THEN CAST('true' AS CHAR) E' at line 1] with root cause
com.mysql.jdbc.exceptions.jdbc4.MySQLSyntaxErrorException: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'WHERE (CAST((CASE WHEN NULL IS NULL AND NULL IS NULL THEN CAST('true' AS CHAR) E' at line 1
    at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[na:1.7.0_80]
    at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:57) ~[na:1.7.0_80]
    at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[na:1.7.0_80]
    at java.lang.reflect.Constructor.newInstance(Constructor.java:526) ~[na:1.7.0_80]
    at com.mysql.jdbc.Util.handleNewInstance(Util.java:400) ~[mysql-connector-java-5.1.36.jar:5.1.36]
    at com.mysql.jdbc.Util.getInstance(Util.java:383) ~[mysql-connector-java-5.1.36.jar:5.1.36]
    at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:980) ~[mysql-connector-java-5.1.36.jar:5.1.36]
    at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3847) ~[mysql-connector-java-5.1.36.jar:5.1.36]
    at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3783) ~[mysql-connector-java-5.1.36.jar:5.1.36]
    at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:2447) ~[mysql-connector-java-5.1.36.jar:5.1.36]
    at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2594) ~[mysql-connector-java-5.1.36.jar:5.1.36]
    at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2541) ~[mysql-connector-java-5.1.36.jar:5.1.36]
    at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2499) ~[mysql-connector-java-5.1.36.jar:5.1.36]
    at com.mysql.jdbc.StatementImpl.executeQuery(StatementImpl.java:1432) ~[mysql-connector-java-5.1.36.jar:5.1.36]
    at com.jolbox.bonecp.StatementHandle.executeQuery(StatementHandle.java:464) ~[bonecp-0.8.0.RELEASE.jar:na]
    at org.aksw.sparqlmap.core.db.DBAccess.executeSQL(DBAccess.java:62) ~[bin/:na]
    at org.aksw.sparqlmap.core.SparqlMap.executeSelect(SparqlMap.java:450) ~[bin/:na]
    at org.aksw.sparqlmap.core.SparqlMap.executeSelect(SparqlMap.java:437) ~[bin/:na]
    ...
```

The SPARQL query was:

```
SELECT DISTINCT  ?o
FROM <http://thesimpsons.com/GraphLisa>
FROM <http://thesimpsons.com/GraphSimpsons>
FROM <http://thesimpsons.com/GraphChild>
FROM <http://thesimpsons.com/GraphPublic>
FROM NAMED <http://thesimpsons.com/GraphLisa>
FROM NAMED <http://thesimpsons.com/GraphSimpsons>
FROM NAMED <http://thesimpsons.com/GraphChild>
FROM NAMED <http://thesimpsons.com/GraphPublic>
WHERE
  { <http://thesimpsons.com/Bart> <http://xmlns.com/foaf/0.1/nick> ?o }
ORDER BY ?o
```

Using the debugger I found the query generating the error:

```
SELECT DISTINCT CAST(NULL AS DECIMAL) "o_R2R_01TP", CAST(NULL AS CHAR) "o_R2R_02LITT", CAST(NULL AS CHAR) "o_R2R_03LITL", CAST(NULL AS CHAR) "o_R2R_04LS", CAST(NULL AS DECIMAL) "o_R2R_05LN", CAST(NULL AS DATETIME) "o_R2R_06LD", CAST(NULL AS CHAR) "o_R2R_07LB", CAST(NULL AS BINARY) "o_R2R_08_LZ"  WHERE (CAST((CASE WHEN NULL IS NULL AND NULL IS NULL THEN CAST('true' AS CHAR) ELSE CAST(NULL AS BINARY) = CAST(NULL AS BINARY) END) AND (CASE WHEN NULL IS NULL AND NULL IS NULL THEN CAST('true' AS CHAR) ELSE CAST(NULL AS CHAR) = CAST(NULL AS CHAR) END) AND (CASE WHEN NULL IS NULL AND NULL IS NULL THEN CAST('true' AS CHAR) ELSE CAST(NULL AS DATETIME) = CAST(NULL AS DATETIME) END) AND (CASE WHEN NULL IS NULL AND NULL IS NULL THEN CAST('true' AS CHAR) ELSE CAST(NULL AS DECIMAL) = CAST(NULL AS DECIMAL) END) AND (CASE WHEN NULL IS NULL AND NULL IS NULL THEN CAST('true' AS CHAR) ELSE CAST(NULL AS CHAR) = CAST(NULL AS CHAR) END) AND (CASE WHEN 'urn:x-arq:DefaultGraphNode' IS NULL AND NULL IS NULL THEN CAST('true' AS CHAR) ELSE NULL = CAST('urn:x-arq:DefaultGraphNode' AS CHAR) END) AS CHAR)) AND (CAST((CASE WHEN NULL IS NULL AND NULL IS NULL THEN CAST('true' AS CHAR) ELSE CAST(NULL AS BINARY) = CAST(NULL AS BINARY) END) AND (CASE WHEN NULL IS NULL AND NULL IS NULL THEN CAST('true' AS CHAR) ELSE CAST(NULL AS CHAR) = CAST(NULL AS CHAR) END) AND (CASE WHEN NULL IS NULL AND NULL IS NULL THEN CAST('true' AS CHAR) ELSE CAST(NULL AS DATETIME) = CAST(NULL AS DATETIME) END) AND (CASE WHEN NULL IS NULL AND NULL IS NULL THEN CAST('true' AS CHAR) ELSE CAST(NULL AS DECIMAL) = CAST(NULL AS DECIMAL) END) AND (CASE WHEN NULL IS NULL AND NULL IS NULL THEN CAST('true' AS CHAR) ELSE CAST(NULL AS CHAR) = CAST(NULL AS CHAR) END) AND (CASE WHEN 'http://thesimpsons.com/Bart' IS NULL AND NULL IS NULL THEN CAST('true' AS CHAR) ELSE NULL = CAST('http://thesimpsons.com/Bart' AS CHAR) END) AS CHAR)) AND (CAST((CASE WHEN NULL IS NULL AND NULL IS NULL THEN CAST('true' AS CHAR) ELSE CAST(NULL AS BINARY) = CAST(NULL AS BINARY) END) AND (CASE WHEN NULL IS NULL AND NULL IS NULL THEN CAST('true' AS CHAR) ELSE CAST(NULL AS CHAR) = CAST(NULL AS CHAR) END) AND (CASE WHEN NULL IS NULL AND NULL IS NULL THEN CAST('true' AS CHAR) ELSE CAST(NULL AS DATETIME) = CAST(NULL AS DATETIME) END) AND (CASE WHEN NULL IS NULL AND NULL IS NULL THEN CAST('true' AS CHAR) ELSE CAST(NULL AS DECIMAL) = CAST(NULL AS DECIMAL) END) AND (CASE WHEN NULL IS NULL AND NULL IS NULL THEN CAST('true' AS CHAR) ELSE CAST(NULL AS CHAR) = CAST(NULL AS CHAR) END) AND (CASE WHEN 'http://xmlns.com/foaf/0.1/nick' IS NULL AND NULL IS NULL THEN CAST('true' AS CHAR) ELSE NULL = CAST('http://xmlns.com/foaf/0.1/nick' AS CHAR) END) AS CHAR)) ORDER BY 
```
# Environment
- MySQL server version: 5.5.43-0ubuntu0.14.04.1
- Java mysql connector version: 5.1.36
- jdbc connection URL: `jdbc:mysql://localhost:3306/test?padCharsWithSpace=true&sessionVariables=sql_mode=ANSI_QUOTES&user=user&password=pass`

Mapping:

```
@prefix rr: <http://www.w3.org/ns/r2rml#> .
@prefix simpsons: <http://thesimpsons.com/> .

<#TriplesMapGraffityArtirst>
    rr:logicalTable [ rr:tableName "GraffityArtist" ];
    rr:subjectMap [
        rr:template "http://thesimpsons.com/{personId}";
        rr:class simpsons:GraffityArtist;
    ].
```

Database: 

```
CREATE TABLE "GraffityArtist" (
 personId VARCHAR(200),
 FOREIGN KEY (personId) REFERENCES Person (id) ON DELETE SET NULL
 );
INSERT INTO "GraffityArtist" (personId) VALUES ('Bart');
```

**Comment by tomatophantastico** [comment]
This is a problem with SPARQL queries that are known to not produce any binding during the mapping phase.

**Comment by tomatophantastico** [comment]
fixed in e172721
